semantic/src/test/kotlin/tools/samt/semantic/SemanticCheckerTest.kt

```diff
@@ -245,16 +245,16 @@ class SemanticCheckerTest {
                 package complex
 
                 record Complex {
-                    int: Int (range(1 .. 1.5))
```

Comment:
Maybe add another test that explicitly tests for Int..Float, Float..Int ranges to work?